near-sdk/src/types/gas.rs

```diff
@@ -56,6 +66,24 @@ impl From<u64> for Gas {
     }
 }
 
+fn isNum(c: char) -> bool {
+  match c {
+    '0'..='9' => true,
+    _ => false
+  }
+}
+
+impl FromStr for Gas {
+  type Err = ParseIntError;
```

Comment:
Also, might be worth, if we are going to have this on the SDK size, to have a custom error for this to not be tied to the `std` num parsing error? I am not convinced the SDK needs `FromStr` for gas, but I'm open to hearing a use case

Reply:
Yeah I just started this PR with the intent of adding the tgas as that is a common thing I see across rust projects.
 
Been updating several to use the new API and I am doing this a lot:
e.g. 
```rs
const GAS_REQURIED: Gas = Gas(500000000000000)
```

Which just doesn't read well as it is unclear how much terragas that is.   The from string is more of a pattern I see on the JS side with testing and apps.  They use a string to represent the u64 as that is what is it is represented as in JSON. 

So I thought perhaps it would be worth allowing `_`'s coming from JS since it makes things more readable.  But I'm also not tied at all to this and have a been working on making a better `tGas` function on the JS side that will prevent the need for `_`'s. So I'll strip this from the PR to keep it simple.

Reply:
But this `FromStr` would only be used from the rust runner, not the js one, no? All I'm suggesting is that this parsing is done on this external usage or if you're using it outside of the SDK in rust you can just parse to u64 and wrap manually or through some helper function outside of the SDK.

Just hard to rationalize keeping this `FromStr` impl for a long-term maintainability perspective since there is no use within the SDK for this.

Reply:
Why the but? I said I'd strip it from this PR as it should be done externally.